src/Microsoft.FeatureManagement/FeatureManager.cs

```diff
@@ -54,14 +65,106 @@ public Task<bool> IsEnabledAsync<TContext>(string feature, TContext appContext,
+        }
+
+        private async ValueTask<T> GetVariantAsync<T, TContext>(string feature, TContext appContext, bool useAppContext, CancellationToken cancellationToken)
+        {
+            if (feature == null)
+            {
+                throw new ArgumentNullException(nameof(feature));
+            }
+
+            FeatureVariant variant = null;
+
+            FeatureDefinition featureDefinition = await _featureDefinitionProvider.GetFeatureDefinitionAsync(feature, cancellationToken).ConfigureAwait(false);
+
+            if (featureDefinition != null)
+            {
+                if (string.IsNullOrEmpty(featureDefinition.Assigner))
+                {
+                    throw new FeatureManagementException(
+                        FeatureManagementError.InvalidConfiguration,
+                        $"Invalid feature variant assigner name for the feature {feature}");
+                }
+
+                IFeatureVariantAssignerMetadata assigner = GetFeatureAssignerMetadata(featureDefinition.Assigner);
+
+                if (assigner == null)
+                {
+                    string errorMessage = $"The feature assigner '{featureDefinition.Assigner}' specified for feature '{feature}' was not found.";
+
+                    if (!_options.IgnoreMissingFeatureAssigners)
+                    {
+                        throw new FeatureManagementException(FeatureManagementError.MissingFeatureAssigner, errorMessage);
+                    }
+                    else
+                    {
+                        _logger.LogWarning(errorMessage);
+
+                        return default(T);
+                    }
+                }
+
+                var context = new FeatureVariantAssignmentContext()
+                {
+                    FeatureDefinition = featureDefinition
+                };
+
+                //
+                // IContextualFeatureVariantAssigner
+                if (useAppContext)
+                {
+                    ContextualFeatureVariantAssignerEvaluator contextualAssigner = GetContextualFeatureVariantAssigner(featureDefinition.Assigner, typeof(TContext));
+
+                    if (contextualAssigner != null)
+                    {
+                        variant = await contextualAssigner.AssignVariantAsync(context, appContext, cancellationToken).ConfigureAwait(false);
+                    }
+                }
+
+                //
+                // IFeatureVariantAssigner
+                if (assigner is IFeatureVariantAssigner featureVariantAssigner)
+                {
+                    variant = await featureVariantAssigner.AssignVariantAsync(context, cancellationToken).ConfigureAwait(false);
+                }
+            
+                if (variant == null &&
+                    featureDefinition.Variants != null)
+                {
+                    variant = featureDefinition.Variants.FirstOrDefault(v => v.Default);
```

Comment:
>FirstOrDefault [](http://example.com/codeflow?start=57&length=14)

If multiple variants contain `Default=true`, we will pick the first variant. But in general, .NET configuration lets the last value win. Should that pattern apply here too?

Reply:
Another option is to throw if there are multiple defaults. I suppose that would be an unexpected state to be in.

Reply:
+1 for throw

Reply:
Done.